File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.js
/**
 * @param {string} s
 * @return {boolean}
 */
const halvesAreAlike = s => {
  let sum = 0;
  const vowels = new Set(['A', 'E', 'I', 'O', 'U', 'a', 'e', 'i', 'o', 'u']);
  for (let i = 0; i < s.length / 2; i++){
     sum += vowels.has(s[i]) - vowels.has(s[s.length - 1 - i]); 
  }
  return !sum;
};

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        mid = len(s) // 2
        a = s[:mid]
        b = s[mid:]
        
        regex = re.compile(r'[aeiouAEIOU]', re.IGNORECASE)
        
        a_count = sum(1 for char in a if regex.match(char))
        b_count = sum(1 for char in b if regex.match(char))
        
        return a_count == b_count